docs/design2.md

```diff
@@ -0,0 +1,109 @@
+# Key Transparency 2.0
+
+## Objective
+
+Key Transparency currently suffers from a design-space trade-off in two dimensions between
+low latency key updates and verification bandwidth; each snapshot of the
+key-value Merkle tree creates an additional point in time that clients must
+audit when inspecting their account history.
+Using Key Transparency v0.4.0 and below requires the presence of powerful
+auditors who can verify that no information has been lost between all snapshots.
+
+| Update Latency | History Audit    | Key Lookup |
+|----------------|------------------|------------|
+| hr             | 45Kb/day         | ~2Kb       |
+| min            | 2Mb/day          | ~2Kb       |
+| s              | 150Mb/day        | ~2Kb       |
+
+
+Key Transparency 2.0 is a new set of algorithms that removes this fundamental trade-off by sharing
+the task of auditing between pairs of clients. As a result, all configurations have efficiently
+auditable data structures that no not require the presence of 3rd party auditors.
+
+We expect that many applications will want to prioritize lower latency and
+reliability by using a *trust, then verify* mode whereby clients receive key
+updates that they can later verify as being part of the global, consistent
+state.  However, this does not preclude individual clients that have a lower
+tolerance for risk from waiting for full, globally consistent verification
+before using fresh public keys from their peers.
+
+| Update Latency | History Audit    | Key Lookup |
+|----------------|------------------|------------|
+| hr             | 18Kb O(1)        | ~2Kb       |
```

Comment:
What does O(1) refer to? Aren't we talking about O(log N) inclusion checks here?

Reply:
`O(log T * log N)`